sql/catalyst/src/main/scala/org/apache/spark/sql/types/Metadata.scala

```diff
@@ -215,6 +215,8 @@ object Metadata {
         x.##
       case x: Metadata =>
         hash(x.map)
+      case null =>
```

Comment:
Basically, we do not support `null` in Metadata.

Reply:
nvm, I found we added the support in https://github.com/apache/spark/pull/8969

Reply:
Yes, the `putNull` method exists, but is actually not useable because of this bug. Having `null` in Metadata would be really helpful for me, as I'd like to store default values for fields in the Metadata. And NULL is a very common case as a default value.